scene/main/viewport.h

```diff
@@ -264,6 +264,9 @@ class Viewport : public Node {
 	Transform3D physics_last_camera_transform;
 	ObjectID physics_last_id;
 
+	uint32_t input_cull_mask_2d = 0xffffffff; // By default, all collision layers are pickable
+	uint32_t input_cull_mask_3d = 0xffffffff;
```

Comment:
I would think, that a single cull mask should be sufficient, that is valid for both, 2d and 3d.
Most scenes are exclusively either 2d or 3d, so this could be adequate.

Reply:
The reason I didn't do this is because every setter for this cull mask, notably the graphical ones in the editor, require a hint that specifies whether to show 2D or 3D collision layers.
Not adhering to that in certain situations is probably highly confusing to potential users.
Making a combined 2D/3D cull mask would, IMO, require the addition of a new hint that either switches what layers it shows contextually, or one that explicitly does not show specific layers.

Reply:
2D and 3D in Godot are whole different separate beasts. They do not often intertwine, so that would be an exception to the norm.